tests/testTriggers.go

```diff
@@ -1375,6 +1627,308 @@ func TriggerRestartVolDriver(contexts *[]*scheduler.Context, recordChan *chan *E
+					err = Inst().V.EnterPoolMaintenance(appNode)
+					if err != nil {
+						UpdateOutcome(event, err)
+						return
+					}
+
+				})
+			stepLog = "wait for 15 mins for apps and volumes to reallocate"
+			Step(stepLog, func() {
+				log.InfoD(stepLog)
+				time.Sleep(15 * time.Minute)
+				validateContexts(event, contexts)
+
+			})
+			stepLog = fmt.Sprintf("exit pool maintenance on node %s", appNode.Name)
+			Step(stepLog,
+				func() {
+					log.InfoD(stepLog)
+					taskStep := fmt.Sprintf("exit pool maintenance on node: %s.",
+						appNode.Name)
+					event.Event.Type += "<br>" + taskStep
+					err = Inst().V.ExitPoolMaintenance(appNode)
+					if err != nil {
+						UpdateOutcome(event, err)
+						return
+					}
+				})
+
+			Step("Giving few seconds for volume driver to stabilize", func() {
+				time.Sleep(20 * time.Second)
+			})
+			err = ValidateDataIntegrity(&nodeContexts)
+			UpdateOutcome(event, err)
+
+			validateContexts(event, contexts)
+		}
+		updateMetrics(*event)
+	})
+}
+
+func isNodeHealthy(n node.Node, eventType string) error {
+	status, err := Inst().V.GetNodeStatus(n)
+	if err != nil {
+		log.Errorf("Unable to get Node [%s] status, skipping [%s] for the node", n.Name, eventType)
+		return err
+	}
+
+	if *status != opsapi.Status_STATUS_OK {
+		log.Errorf("Node [%s] is not healthy,  skipping [%s] for the node", n.Name, eventType)
+		return fmt.Errorf("node [%s] has status [%v]", n.Name, status)
+	}
+	return nil
+}
+
+// TriggerStorageFullPoolExpansion performs pool expansion on storageFull nodes
+func TriggerStorageFullPoolExpansion(contexts *[]*scheduler.Context, recordChan *chan *EventRecord) {
+	defer ginkgo.GinkgoRecover()
+	defer endLongevityTest()
+	startLongevityTest(StorageFullPoolExpansion)
+	event := &EventRecord{
+		Event: Event{
+			ID:   GenerateUUID(),
+			Type: StorageFullPoolExpansion,
+		},
+		Start:   time.Now().Format(time.RFC1123),
+		Outcome: []error{},
+	}
+
+	defer func() {
+		event.End = time.Now().Format(time.RFC1123)
+		*recordChan <- event
+	}()
+	setMetrics(*event)
+	stepLog := "get nodes and expand storagefull pools"
+	Step(stepLog, func() {
+		log.InfoD(stepLog)
+		for _, appNode := range node.GetStorageNodes() {
+
+			err := isNodeHealthy(appNode, event.Event.Type)
+			if err != nil {
+				UpdateOutcome(event, err)
+				continue
+			}
+			poolsStatus, err := Inst().V.GetNodePoolsStatus(appNode)
+			if err != nil {
+				UpdateOutcome(event, err)
+				continue
+			}
+			for poolUUID, v := range poolsStatus {
+				if v == "Offline" {
+					taskStep := fmt.Sprintf("expanding storagefull pool [%s] on node [%s]", poolUUID,
+						appNode.Name)
+					event.Event.Type += "<br>" + taskStep
+					log.InfoD("Pool [%s] on node [%s] is offline", v, appNode.Name)
+					poolResizeType := []opsapi.SdkStoragePool_ResizeOperationType{opsapi.SdkStoragePool_RESIZE_TYPE_AUTO,
+						opsapi.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK}
+
+					//Remove it once SdkStoragePool_RESIZE_TYPE_ADD_DISK is supported in dmthin
```

Comment:
// TODO: Remove it once SdkStoragePool_RESIZE_TYPE_ADD_DISK is supported in dmthin?